src/FSharpLint.Core/Rules/Conventions/Naming/UnnestedFunctionNames.fs

```diff
@@ -0,0 +1,25 @@
+module FSharpLint.Rules.UnnestedFunctionNames
+
+open FSharp.Compiler.Syntax
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.AstInfo
+open FSharpLint.Framework.Rules
+open FSharpLint.Rules.Helper.Naming
+
+let private getIdentifiers (args: AstNodeRuleParams) =
+    match args.AstNode with
+    | AstNode.Binding (SynBinding (_, _, _, _, _attributes, _, _, pattern, _, _, _, _)) ->
+        if isNested args args.NodeIndex then
+            Array.empty
+        else
+            getPatternIdents Accessibility.Public getFunctionIdents true pattern
```

Comment:
are we sure we want to use Accessibility.Public here? we want all functions, even private and internal ones

Reply:
I don't think the `Accessibility.Public` is used in this case. You think it's best to add a `Both` or `NotApplicable` member to the `Accessibility` DU?

Reply:
I don't understand how can this be not used?

Reply:
Passing `Accessibility.Public` here means we start within public scope.
So it's `getFunctionIdents` function that can decide what is included based on identifier's accessibility.
Which ignores accessibility parameter, and processes all functions.